pkg/subctl/cmd/utils/restconfig/restconfig.go

```diff
@@ -0,0 +1,162 @@
+				return nil, err
+			}
+
+			restConfigs = append(restConfigs, config)
+		}
+	}
+
+	return restConfigs, nil
+}
+
+func GetBrokerRestConfigAndNamespace(submariner *v1alpha1.Submariner,
+	serviceDisc *v1alpha1.ServiceDiscovery) (*rest.Config, string, error) {
+	if submariner != nil {
+		// Try to authorize against the submariner Cluster resource as we know the CRD should exist and the credentials
+		// should allow read access.
+		restConfig, _, err := resource.GetAuthorizedRestConfig(submariner.Spec.BrokerK8sApiServer, submariner.Spec.BrokerK8sApiServerToken,
+			submariner.Spec.BrokerK8sCA, rest.TLSClientConfig{}, schema.GroupVersionResource{
+				Group:    subv1.SchemeGroupVersion.Group,
+				Version:  subv1.SchemeGroupVersion.Version,
+				Resource: "clusters",
+			}, submariner.Spec.BrokerK8sRemoteNamespace)
+
+		return restConfig, submariner.Spec.BrokerK8sRemoteNamespace, err
+	}
+
+	if serviceDisc != nil {
+		// Try to authorize against the ServiceImport resource as we know the CRD should exist and the credentials
+		// should allow read access.
+		restConfig, _, err := resource.GetAuthorizedRestConfig(serviceDisc.Spec.BrokerK8sApiServer, serviceDisc.Spec.BrokerK8sApiServerToken,
+			serviceDisc.Spec.BrokerK8sCA, rest.TLSClientConfig{}, schema.GroupVersionResource{
+				Group:    "multicluster.x-k8s.io",
+				Version:  "v1alpha1",
+				Resource: "serviceimports",
+			}, serviceDisc.Spec.BrokerK8sRemoteNamespace)
+
+		return restConfig, serviceDisc.Spec.BrokerK8sRemoteNamespace, err
+	}
+
+	return nil, "", nil
+}
+
+func GetClientConfigAndClusterName(rules *clientcmd.ClientConfigLoadingRules, overrides *clientcmd.ConfigOverrides) (RestConfig, error) {
+	config := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(rules, overrides)
+	clientConfig, err := config.ClientConfig()
+	if err != nil {
+		return RestConfig{}, err
+	}
+
+	raw, err := config.RawConfig()
+	if err != nil {
+		return RestConfig{}, err
+	}
+
+	clusterName := GetClusterNameFromContext(raw, overrides.CurrentContext)
+
+	if clusterName == nil {
+		return RestConfig{}, fmt.Errorf("could not obtain the cluster name from kube config: %#v", raw)
+	}
+
+	return RestConfig{Config: clientConfig, ClusterName: *clusterName}, nil
+}
+
+func GetClients(config *rest.Config) (dynamic.Interface, kubernetes.Interface, error) {
+	dynClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return nil, nil, err
+	}
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, nil, err
+	}
+	return dynClient, clientSet, nil
+}
+
+func GetClusterNameFromContext(rawConfig api.Config, overridesContext string) *string {
+	if overridesContext == "" {
+		// No context provided, use the current context
+		overridesContext = rawConfig.CurrentContext
+	}
+	configContext, ok := rawConfig.Contexts[overridesContext]
+	if !ok {
+		return nil
+	}
+	return &configContext.Cluster
+}
+
+func GetRestConfig(kubeConfigPath, kubeContext string) (*rest.Config, error) {
```

Comment:
```suggestion
func ForCluster(kubeConfigPath, kubeContext string) (*rest.Config, error) {
```

Would look better `restconfig.ForCluster`.
Also the module is already called `restconfig` so repeating it here is stuttering :)